### PR TITLE
[SPARK-5865] [API DOC] Add doc warnings for methods that return local data structures

### PR DESCRIPTION
**Pull request by Wenpei**
@rxin @srowen 
I work out note message for rdd.take function, please help to review.

If it's fine, I can apply to all other function later.

### CODE REVIEW

**Comment by srowen** [comment]
I think it's OK, though I might say something like "all the data is loaded into the driver's memory" rather than "the whole _thing_ ..." Very minor preference.

**Comment by Wenpei** [comment]
@srowen Finish task. pls help to review and launch test.

**Comment by srowen** [comment]
Looking better. I'm wondering whether it's useful to also copy this to `takeOrdered`, `takeSample`, `top` which also return arrays. They're less used. You could also use `@note` syntax in scaladoc.

**Comment by Wenpei** [comment]
@srowen  Done

**Comment by srowen** [comment]
LGTM. I'll leave it open a day for comments.

File: python/pyspark/sql/dataframe.py [hunk 1]
```diff
@@ -1330,6 +1333,9 @@ def toDF(self, *cols):
     def toPandas(self):
         """Returns the contents of this :class:`DataFrame` as Pandas ``pandas.DataFrame``.
 
+        Note that this method should only be used if the resulting array is expected
```
Comment (holdenk) [inline python/pyspark/sql/dataframe.py]: this is super minor, but toPandas results in a Panda's DataFrame rather than an array

**Comment by holdenk** [comment]
Since we are doing the Scala & Python APIs at the same time might it make sense to update the Java API at the same time? Also possibly the similar collect functions on PairRDD functions (collectAsMap)?

**Comment by Wenpei** [comment]
@holdenk I add note for Java API, and change "array" to "Pandas's DataFrame" for toPandas function, pls review this change. :-)

**Comment by srowen** [comment]
Thanks @Wenpei I think it's really useful to thoroughly document this while we're here. This is pretty complete now, though I think we're missing the Scala collectAsMap? how about the Python version? That may be it then.

**Comment by Wenpei** [comment]
@srowen done for scala and python's  collectAsMap

**Comment by srowen** [comment]
Jenkins, retest this please

**Comment by Wenpei** [comment]
Jenkins, retest this please

**Comment by srowen** [comment]
Jenkins, retest this please

**Comment by Wenpei** [comment]
@srowen Can you launch test again,  not sure why but get exception below:
java.lang.NullPointerException
	at org.apache.spark.SparkFiles$.getRootDirectory(SparkFiles.scala:37)

**Comment by Wenpei** [comment]
@srowen @SparkQA Can you launch test again, doc change shouldn't break spark unit test.

**Comment by Wenpei** [comment]
ping @srowen @rxin

**Comment by srowen** [comment]
Jenkins, retest this please

**Comment by Wenpei** [comment]
@srowen ready to merge

**Comment by srowen** [comment]
Merged to master

**Comment by srowen** [comment]
@Wenpei if you have a JIRA handle I'll assign it to you

**Comment by Wenpei** [comment]
@srowen i add a comment in jira ,  which user name is Tommy Yu.

**Pull request closed without merging**